lib/DDG/Meta/Data.pm

```diff
@@ -38,14 +38,16 @@ unless(%ia_metadata){
     unless($ENV{NO_METADATA_DOWNLOAD}){
         my $ua = LWP::UserAgent->new;
         $ua->timeout(5);
-        #$ua->default_header('Accept-Encoding' => scalar HTTP::Message::decodable());
-        my $res = $ua->mirror("https://duck.co/ia/repo/all/json", $f);
+        $ua->default_header('Accept-Encoding' => scalar HTTP::Message::decodable());
+        my $res = $ua->mirror('https://duck.co/ia/repo/all/json', $f);
         unless($res->is_success || $res->code == 304){
             debug && warn "Failed to download metdata: " . $res->status_line;
         }
     }
 
-    my $metadata = decode_json(io($f)->all);
+    open my $fh, -B $f ? '<:gzip' : '<', $f or die "Failed to open file $f: $!";
```

Comment:
The -B confused me in the ternary at first. Then the `do { local $/; <$fh> }` block below.